system/Common.php

```diff
@@ -403,14 +400,14 @@ function force_https(int $duration = 31536000, RequestInterface $request = null,
 			$response = Services::response(null, true);
 		}
 
-		if (is_cli() || $request->isSecure())
+		if (ENVIRONMENT !== 'testing' && (is_cli() || $request->isSecure()))
 		{
 			return;
 		}
-		// @codeCoverageIgnoreStart
-		// If the session library is loaded, we should regenerate
+
+		// If the session status is active, we should regenerate
 		// the session ID for safety sake.
-		if (class_exists('Session', false))
+		if (session_status() === PHP_SESSION_ACTIVE)
```

Comment:
Also fix the session check as well. Previosly, the `class_exists('Session')` will always return false as no class named `Session`. Changed with session active check `session_status() === PHP_SESSION_ACTIVE` above.